metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -87,13 +88,30 @@ class DebugProvider(
     sourceMapper: SourceMapper,
     userConfig: () => UserConfiguration,
     testProvider: TestSuitesProvider,
-) extends Cancelable {
+) extends Cancelable
+    with LogForwarder {
 
   import DebugProvider._
 
   private val debugSessions = new MutableCancelable()
 
-  override def cancel(): Unit = debugSessions.cancel()
+  private val currentRunner =
+    new ju.concurrent.atomic.AtomicReference[DebugRunner](null)
+
+  override def info(message: String): Unit = {
+    val runner = currentRunner.get()
+    if (runner != null) runner.stdout(message)
+  }
+
+  override def error(message: String): Unit = {
+    val runner = currentRunner.get()
+    if (runner != null) runner.error(message)
+  }
+
+  override def cancel(): Unit = {
+    Option(currentRunner.get()).foreach(_.cancel())
```

Comment:
consistency nit: in info and error you have explicit null guard, here you're relying on Option constructor to handle it. I find `if (runner != null)` more suitable in this case as we're on low level anyway.

Reply:
Fixed